### DISCUSSION TITLE
Wait for rotary encoder button to be pushed to exit COMMAND

### DISCUSSION THREAD

**Discussion by B1ackt34** [forntoh/LcdMenu#267] [Q&A] [answered]
Hello, first i want to thank you for this very good library !

I'm working on a project with several commands.

One command needs to hide menu (ok), execute a command and wait until the user click the button of the encoder.
I've searched in other discussions but i can't find any command to use.
I guess i need to do a WHILE ... function waiting for the button to be pressed.
Does exist any command i can put there?

Thank you in advance.

**Comment by B1ackt34** [comment, marked as answer]
After doing a bit of research I answer myself.
You can use

`encoder.push()`

**Reply by kimbrussow** [reply]
does the encoder.push() help you get out of a do.....while loop?

**Reply by B1ackt34** [reply]
Yes, here's the code

```
void ConfigPortalVoid () {
  lcd.clear();
  lcd.setCursor(0, 0);
  lcd.print("Configuration portal");
  lcd.setCursor(3, 1);
  lcd.print("is active at");
  lcd.setCursor(3, 2);
  if (WiFi.status() == WL_CONNECTED) {
    lcd.print(WiFi.localIP());
  } else {
    lcd.print(WiFi.softAPIP());
  }
  lcd.setCursor(0, 3);
  lcd.print("Press button to exit");
  delay(500);
  Serial.println("Button Pressed, Starting Portal");
  wifiManager.startWebPortal();
  do {
  Serial.println("portal running");
  wifiManager.process();
  } while (encoder.push() == 0);
  Serial.println("pulsante premuto");
  wifiManager.stopWebPortal();
  }
```

**Reply by kimbrussow** [reply]
Thanks for the info and pointing to the right area. I am running a led light strip that uses a lot of delays. I added my own delay to use your thoughts:
bool ntDelay(int period)
{
 bool retValue = true;
 time_target = millis()+ period ; 
 do{
      if (encoder.rotate()!= 0)
        {
          retValue = false;
          break;}
       if (encoder.push()!= 0)
        {
          retValue = false;
          break;}  
 }while(millis() <= time_target) ;
  return retValue;
}
void HappyProgram
{
in the do loop I changed the delays:

do { happy stuff here
blah blah
if (!ntDelay(100))
          {goto bailout;}

 }while (ntDelay(1)== true);  // << not really needed

bailout:
   Serial.println("out of loop");
} // end of HappyProgram

works great thank you!!

**Reply by B1ackt34** [reply]
Happy to have been of help !
